core/src/main/scala/akka/persistence/cassandra/query/EventsByPersistenceIdStage.scala

```diff
@@ -324,11 +329,19 @@ import com.datastax.driver.core.utils.Bytes
               if (empty && switchPartition && lookingForMissingSeqNr.isEmpty) {
                 if (refreshInterval.isEmpty) {
                   completeStage()
+                } else {
+                  if (pendingFastForward.isDefined) {
```

Comment:
`pendingFastForward.foreach`, not in super hot path, and we create plenty of objects in those queries anyway